### PR TITLE
ignore posts which were set as draft in the front-matter

### PR DESCRIPTION
**Pull request by mamboer**

### CODE REVIEW

File: atom.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
     <published>{{ post.date.toISOString() }}</published>
     <updated>{{ post.updated.toISOString() }}</updated>
     {% if config.feed.content and post.content %}
-    <content type="html"><![CDATA[{{ post.content | safe }}]]></content>
+    <content type="html"><![CDATA[{{ post.content | uriencode }}]]></content>
```
Comment (NoahDragon) [inline atom.xml]: Why need to change this?

Comment (mamboer) [inline atom.xml]: It sucks when the post's content contains emoji chars like 😀
Use the `uriencode` filter can resolve this problem.
https://github.com/o2team/o2team.github.io/issues/21

Comment (NoahDragon) [inline atom.xml]: Thanks for the clarification.

**Comment by NoahDragon** [comment]
LGTM. Could you please fix the jscs error?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "hexo-generator-feed",
-  "version": "1.2.0",
+  "name": "hexo-generator-xfeed",
```
Comment (NoahDragon) [inline package.json]: I notice the package name has changed. Is it a typo?

File: package.json [hunk 3]
```diff
@@ -23,18 +23,18 @@
...
   },
   "devDependencies": {
     "chai": "^3.5.0",
-    "cheerio": "^0.20.0",
-    "eslint": "^2.4.0",
-    "eslint-config-hexo": "^1.0.3",
-    "hexo": "^3.0.0",
-    "jscs": "^2.11.0",
+    "cheerio": "^0.22.0",
+    "eslint": "^3.19.0",
```
Comment (NoahDragon) [inline package.json]: I suggest keeping these package in their old version. I assume the CI errors are caused by the new versions.

**Comment by NoahDragon** [comment]
Thanks for your contribution.

**Pull request merged**